test/metabase/test/redefs.clj

```diff
@@ -29,6 +62,13 @@
         (next-method model attributes f)))
     (next-method model attributes f)))
 
+(defmacro with-temp-persisted!
+  "Like [[mt/with-temp]], but all the created items are persisted after."
+  [& args]
+  `(binding [*with-temp-clean-up*           false
+             tu.thread-local/*thread-local* true]
```

Comment:
Can you explain why `tu.thread-local/*thread-local*` is set to `true`?

Reply:
It should be `false`! I need to set this so we don't run `with-temp` in a transaction.

I think the reason why test passed is because `{:rollback-only true}` from `t2/with-transaction` does not actually works :)